code/datums/components/transfers.dm

```diff
@@ -104,7 +104,14 @@
 
 /datum/component/transfer_input/proc/handle_incoming(comsig_target, atom/movable/incoming)
 	if(is_permitted(incoming))
-		incoming.set_loc(parent)
+		if (istype(incoming, /obj/item))
+			var/obj/item/I = incoming
+			if (I.stored)
+				I.stored.transfer_stored_item(I, parent)
+			else
+				I.set_loc(parent)
```

Comment:
No reason, I think that could be done

Reply:
Not going to worry about it for now 👍

Reply:
Thinking about it more, it was as far as I remembered there wasn't a reason, but there's also the issue of that transfer_stored_item() can accept different arguments, so there would be assumptions automatically set in set_loc() that wouldn't catch everything

Reply:
Went ahead and did it, feel like it would be best considering solutions for https://github.com/goonstation/goonstation/issues/13708 and https://github.com/goonstation/goonstation/issues/13714 were this too, should hopefully catch all future stuff